### PR TITLE
Apply `unusedArguments` rule for `swiftformat`

### PR DESCRIPTION
**Pull request by cgrindel**

### CODE REVIEW

File: tools/generator/src/Generator/AddBazelDependenciesTarget.swift [hunk 1]
```diff
@@ -404,9 +404,9 @@ fi
 
     private static func copyFilesCommand(
         generatedInputsFileList: String,
-        hasGeneratedFiles: Bool,
-        files: [FilePath: File],
-        filePathResolver: FilePathResolver
+        hasGeneratedFiles _: Bool,
+        files _: [FilePath: File],
+        filePathResolver _: FilePathResolver
```
Comment (brentleyjones) [inline tools/generator/src/Generator/AddBazelDependenciesTarget.swift:409]: Lets remove these ones instead.

File: tools/generator/src/Generator/AddBazelDependenciesTarget.swift [hunk 2]
```diff
@@ -284,7 +284,7 @@ ln -sfn "$PROJECT_DIR" SRCROOT
     }
 
     private static func bazelBuildCommand(
-        buildMode: BuildMode,
+        buildMode _: BuildMode,
```
Comment (brentleyjones) [inline tools/generator/src/Generator/AddBazelDependenciesTarget.swift:287]: Lets remove this instead.

File: tools/generator/src/Generator/Main.swift [hunk 3]
```diff
@@ -41,7 +41,7 @@ extension Generator {
         let buildMode: BuildMode
     }
 
-    static func parseArguments(_ arguments: [String]) throws -> Arguments {
+    static func parseArguments(_: [String]) throws -> Arguments {
         guard CommandLine.arguments.count == 7 else {
```
Comment (brentleyjones) [inline tools/generator/src/Generator/Main.swift:45]: Let's use the passed in value instead.

```suggestion
    static func parseArguments(_ arguments: [String]) throws -> Arguments {
        guard arguments.count == 7 else {
```

File: tools/generator/src/Generator/SetTargetConfigurations.swift [hunk 4]
```diff
@@ -649,7 +649,7 @@ private let iPhonePlatforms: Set<String> = [
 private extension Dictionary
 where Key == BuildSettingConditional, Value == [String: BuildSetting] {
     func asBuildSettingDictionary(
-        buildMode: BuildMode
+        buildMode _: BuildMode
```
Comment (brentleyjones) [inline tools/generator/src/Generator/SetTargetConfigurations.swift:652]: Lets remove this one instead.

File: tools/generator/test/Fixtures.swift [hunk 5]
```diff
@@ -1400,7 +1400,7 @@ done < "$SCRIPT_INPUT_FILE_LIST_0"
         disambiguatedTargets: DisambiguatedTargets,
         files: [FilePath: File],
         products: Products,
-        filePathResolver: FilePathResolver,
+        filePathResolver _: FilePathResolver,
```
Comment (brentleyjones) [inline tools/generator/test/Fixtures.swift:1403]: Let's remove this one instead.

File: tools/generator/src/Generator/WriteXcodeProj.swift [hunk 6]
```diff
@@ -5,7 +5,7 @@ extension Generator {
     /// Writes the ".xcodeproj" file to disk.
     static func writeXcodeProj(
         _ xcodeProj: XcodeProj,
-        buildMode: BuildMode,
+        buildMode _: BuildMode,
```
Comment (brentleyjones) [inline tools/generator/src/Generator/WriteXcodeProj.swift:8]: Let's remove this one instead.

File: tools/generator/src/Generator/SetTargetConfigurations.swift (hunk 4 above)
Comment (cgrindel) [reply tools/generator/src/Generator/SetTargetConfigurations.swift]: Done

File: tools/generator/test/Fixtures.swift (hunk 5 above)
Comment (cgrindel) [reply tools/generator/test/Fixtures.swift]: Done

File: tools/generator/src/Generator/WriteXcodeProj.swift (hunk 6 above)
Comment (cgrindel) [reply tools/generator/src/Generator/WriteXcodeProj.swift]: Done

File: tools/generator/src/Generator/Main.swift [hunk 7]
```diff
@@ -47,7 +47,7 @@ extension Generator {
     }
 
     static func parseArguments(_ arguments: [String]) throws -> Arguments {
-        guard CommandLine.arguments.count == 8 else {
+        guard arguments.count == 8 else {
```
Comment (brentleyjones) [inline tools/generator/src/Generator/Main.swift:50]: Lets replace it everywhere in this functions, not just here.

Comment (cgrindel) [reply tools/generator/src/Generator/Main.swift:50]: Done

**Review by brentleyjones — APPROVED** [review]

**Pull request merged**